### PR TITLE
S2i, build without run. Start named container with command

### PR DESCRIPTION
**Pull request by jfdenise**
This is to address cases where we want to start the s2i built image with a command (eg: bash).
- s2i build only (run False)
- Execute a named container with a cmd.

For example:
Given s2i build <app> from . with env and False using master and run False
When container integ- is started with command bash
...

### CODE REVIEW

**Comment by jfdenise** [comment]
@kabir FYI.

File: steps/s2i_steps.py [hunk 1]
```diff
@@ -52,14 +52,16 @@ def s2i_inner(context, application, path='.', env="", incremental=False, tag="ma
 @given(u's2i build {application} from {path} with env using {tag}')
 @given(u's2i build {application} from {path} with env and {incremental}')
 @given(u's2i build {application} from {path} with env and {incremental} using {tag}')
-def s2i_build(context, application, path='.', env="", incremental=False, tag="master"):
+@given(u's2i build {application} from {path} with env and {incremental} using {tag} and run {run}')
+def s2i_build(context, application, path='.', env="", incremental=False, tag="master", run=True):
```
Comment (goldmann) [inline steps/s2i_steps.py:55]: Wouldn't it better to create a second method that would call this one. Then you could do:

```
@given(u's2i build {application} from {path} with env and {incremental} using {tag} without running'
```
on that method and call it with `run=False`. It seems for me more readable than this. Or just try to reword the `{run}` parameter.

**Review by goldmann — APPROVED** [review]

**Comment by jfdenise** [comment]
Marek, you went too quick, the new method doesn't work. The {tag} capture the without running.

**Comment by jfdenise** [comment]
@goldmann , I am going to name the new method "norun s2i build ..."

**Comment by goldmann** [comment]
Maybe `s2i build and don't run {application}...`?

**Comment by goldmann** [comment]
Or maybe just change the order of definitions, moving the `without running` above. Not sure if this works, but give it a try.

**Comment by jmtd** [comment]
Hi @jfdenise 

> For example:
> Given s2i build from . with env and False using master and run False
> When container integ- is started with command bash

Do you have a link to an example of this pattern in action? I want to do something similar   but I'm not sure how to phrase the second part (container name integ- prefix but I need to estalblish the rest etc)

Thanks!

Edit: I just found something in the eap modules. Thanks anyway!

**Pull request merged**